Mysterious -nan popping up in kernel

The second constant `0.5` in the code

```python
def calculate_velocity_on_wgrid(pyom):
    pyom.u_wgrid[:,:,:-1] = pyom.u[:,:,1:,pyom.tau] * pyom.maskU[:,:,1:] * 0.5 * pyom.dzt[None,None,1:] / pyom.dzw[None,None,:-1] \
                          + pyom.u[:,:,:-1,pyom.tau] * pyom.maskU[:,:,:-1] * 0.5 * pyom.dzt[None,None,:-1] / pyom.dzw[None,None,:-1]
```

is translated to a `-nan` in the JIT kernel:

```
rank: 0, size: 26416, block list:
    rank: 1, size: 49, news: {a49,a190,}, frees: {a49,}, temps: {a49,}, block list:
            BH_MULTIPLY a49[0:26416:49,0:49:1] a112[0:26416:150,1:50:3,1:2:1] a181[0:26416:50,1:50:1]
            BH_MULTIPLY a190[0:26416:49,0:49:1] a49[0:26416:49,0:49:1] 0.5
            BH_FREE a49[0:26416:49,0:49:1]
Write source "/tmp/bohrium_effe/src/d27251f667cc3cdf.c"
rank: 0, size: 26416, block list:
    rank: 1, size: 49, news: {a136,}, frees: {a190,}, block list:
            BH_MULTIPLY a136[0:26416:49,0:49:1] a190[0:26416:49,0:49:1] a154[0:26416:0,1:50:1]
            BH_FREE a190[0:26416:49,0:49:1]
Write source "/tmp/bohrium_effe/src/6a26971750a31219.c"
rank: 0, size: 26416, block list:
    rank: 1, size: 49, news: {a66,a194,a132,a51,}, frees: {a136,a66,a132,a51,}, temps: {a66,a132,a51,}, block list:
            BH_MULTIPLY a132[0:26416:49,0:49:1] a112[0:26416:150,0:49:3,1:2:1] a181[0:26416:50,0:49:1]
            BH_MULTIPLY a51[0:26416:49,0:49:1] a132[0:26416:49,0:49:1] -nan
            BH_FREE a132[0:26416:49,0:49:1]
            BH_DIVIDE a66[0:26416:49,0:49:1] a136[0:26416:49,0:49:1] a53[0:26416:0,0:49:1]
            BH_ADD a194[0:26416:49,0:49:1] a66[0:26416:49,0:49:1] a51[0:26416:49,0:49:1]
            BH_FREE a51[0:26416:49,0:49:1]
            BH_FREE a66[0:26416:49,0:49:1]
            BH_FREE a136[0:26416:49,0:49:1]
Write source "/tmp/bohrium_effe/src/7bfb3319dabe4166.c"
/tmp/bohrium_effe/src/7bfb3319dabe4166.c: In function ‘execute’:
/tmp/bohrium_effe/src/7bfb3319dabe4166.c:18:27: error: wrong type argument to unary minus
                 t3 = t0 * -nan;
```

Unfortunately, I have not been able to reproduce the bug in any other setting. Everything works when I remove either of the arrays, or when I run the code in isolation. The arrays only contain finite values with dtype `float64`. `dzw` does not contain any zeros.

I figured you guys might have an idea what could cause this. Otherwise, I'll have to dig deeper to try and reproduce the problem.

--- a/filter/bccon/contract_collect.cpp
+++ b/filter/bccon/contract_collect.cpp
@@ -186,13 +186,13 @@ void Contracter::contract_collect(bh_ir &bhir)
             for(size_t pc_chain = pc+1; pc_chain < bhir.instr_list.size(); ++pc_chain) {
                 bh_instruction& other_instr = bhir.instr_list[pc_chain];
 
-                if (is_add_sub(collect_opcode) and is_add_sub(other_instr.opcode) and bh_is_constant(&instr.operand[2])) {
+                if (is_add_sub(collect_opcode) and is_add_sub(other_instr.opcode) and bh_is_constant(&other_instr.operand[2])) {
                     // Both are ADD or SUBTRACT
                     if (*views.back() == other_instr.operand[1]) {
                         views.push_back(&other_instr.operand[0]);
                         chain.push_back(&other_instr);
                     }
-                } else if (is_mul_div(collect_opcode) and is_mul_div(other_instr.opcode) and bh_is_constant(&instr.operand[2])) {
+                } else if (is_mul_div(collect_opcode) and is_mul_div(other_instr.opcode) and bh_is_constant(&other_instr.operand[2])) {
                     // Both are MULTIPLY or DIVIDE
 
                     // We are not allowed to DIVIDE when the result operand has integer type

